src/main/java/io/reactiverse/pgclient/impl/PgConnectionImpl.java

```diff
@@ -144,4 +146,32 @@ public void close() {
       context.runOnContext(v -> close());
     }
   }
+
+  @Override
+  public int processId() {
+    return conn.getProcessId();
+  }
+
+  @Override
+  public int secretKey() {
+    return conn.getSecretKey();
+  }
+
+  @Override
+  public PgConnection cancelRequest(Handler<AsyncResult<Void>> handler) {
+    Context current = Vertx.currentContext();
+    if (current == context) {
+      factory.connect(ar -> {
```

Comment:
if I understand correctly, this will initiate the the Postgres protocol before the CancelRequest message is sent, right ?

Reply:
No, as reworked in commit https://github.com/reactiverse/reactive-pg-client/commit/07c8b82ebc4edacbe6543ede266cda7023fc85f8, `connect` method in `PgConnectionFactory` will only initiate a raw tcp connection to server and do SSL information exchange(if enabled), `create` method will actually send a startup message(i.e init Postgres protocol), the code is clear in
https://github.com/reactiverse/reactive-pg-client/blob/07c8b82ebc4edacbe6543ede266cda7023fc85f8/src/main/java/io/reactiverse/pgclient/impl/PgConnectionFactory.java#L92-L102

Reply:
right, I missed that